c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -963,6 +963,15 @@ void* raw_alloc(size_t nbytes) {
   return r;
 }
 
+void* raw_external_alloc(size_t nbytes, cudaStream_t stream) {
```

Comment:
nit: might make sense to call it raw_alloc_with_stream, we might use it internally also.